File: Backend/app/schemas/role.py
from datetime import datetime
from pydantic import BaseModel, EmailStr


class PermissionBase(BaseModel):
    module_name: str
    can_view: bool = True
    can_create: bool = False
    can_update: bool = False
    can_delete: bool = False


class PermissionRead(PermissionBase):
    id: int
    role_id: int

    class Config:
        from_attributes = True


class RoleBase(BaseModel):
    name: str
    description: str | None = None


class RoleCreate(RoleBase):
    pass


class RoleRead(RoleBase):
    id: int
    created_at: datetime
    permissions: list[PermissionRead] = []

    class Config:
        from_attributes = True


class PermissionUpdateItem(BaseModel):
    module_name: str
    can_view: bool
    can_create: bool
    can_update: bool
    can_delete: bool


class RolePermissionsUpdatePayload(BaseModel):
    permissions: list[PermissionUpdateItem]


class StaffUserCreate(BaseModel):
    name: str
    email: EmailStr
    password: str
    role_name: str


class StaffUserRead(BaseModel):
    id: int
    name: str
    email: str
    role: str
    created_at: datetime

    class Config:
        from_attributes = True

File: Frontend/src/types/role.ts
export interface RolePermission {
  id: number;
  role_id: number;
  module_name: string;
  can_view: boolean;
  can_create: boolean;
  can_update: boolean;
  can_delete: boolean;
}

export interface Role {
  id: number;
  name: string;
  description: string;
  created_at: string;
  permissions: RolePermission[];
}

export interface StaffUser {
  id: number;
  name: string;
  email: string;
  role: string;
  created_at: string;
}

export interface RoleCreatePayload {
  name: string;
  description: string;
}

export interface StaffUserCreatePayload {
  name: string;
  email: string;
  password: string;
  role_name: string;
}
